### ISSUE TITLE
[bug: API] Metal Gateway associated with a Vlan does not show up when fetching VirtualNetwork by ID

### ISSUE THREAD

**Issue by rinzler-17** [equinix-labs/metal-java#32]
Sample instance returned by VlansApi.getVirtualNetwork(vlanId)

```
VirtualNetwork {
    assignedTo: class Href {
        href: /metal/v1/projects/{projectId}
        additionalProperties: null
    }
    assignedToVirtualCircuit: false
    description: null
    facility: null
    href: /metal/v1/virtual-networks/{vlanId}
    id: {vlanId}
    instances: []
    metalGateway: null
    metro: class Href {
        href: /metal/v1/locations/metros/{metroId}
        additionalProperties: null
    }
    metroCode: da
    vxlan: 200
    additionalProperties: {metal_gateways={}, created_at={timestamp}, facility_code=null}
}
```

metalGateway is null and metal_gateways under additionalProperties is empty. Also, what is the spec of value keyed by metal_gateways inside additionalProperties?

**Comment by rinzler-17** [comment]
@displague @VamshikShetty

**Comment by displague** [comment]
> metalGateway is null and metal_gateways under additionalProperties is empty
> what is the spec of value keyed by metal_gateways inside additionalProperties

The spec field `metal_gateway` should be `metal_gateways` and should be an array of Metal Gateway objects.

Using the `include=metal_gateways` parameter will expand the metal gateways from an `href` to the full object.

```json
{...
"metal_gateways": [
  {
    "created_at": "2022-05-05T18:08:26Z"
    "gateway_address": "145.40.101.89/29"
    "href": "/metal-gateways/65a7e762-059a-4b98-a08f-55addb729cdf"
    "id": "65a7e762-059a-4b98-a08f-55addb729cdf"
    "state": "active"
    "updated_at": "2022-05-05T18:09:27Z"
    "vlan": 9
  }
]
}
```

**Comment by displague** [comment]
#35 has been merged but I'd like to keep this open issue until we discuss:
* https://github.com/equinix-labs/metal-java/pull/37#discussion_r942621774
* https://github.com/equinix-labs/metal-java/pull/37#discussion_r942624273

**Comment by rinzler-17** [comment]
In response to : 

- https://github.com/equinix-labs/metal-java/pull/37#discussion_r942621774 , I can change the vlan type to integer with range [2-3999] for MetalGatewayLite in another patch. 
- https://github.com/equinix-labs/metal-java/pull/37#discussion_r942624273 , we wanted to know why there is a MetalGatewayLite when there’s already a MetalGateway in the spec and how they're being used.

**Comment by displague** [comment]
> why is there a `MetalGatewayLite` when there’s already a `MetalGateway` in the spec? how are they being used?

The API Spec (and the actual API response) offer a different view of some objects based on the context. 

A “Lite” response (typically those offered as a shallow value with an `href` property, expanded through `?include=foo`) may contain fewer fields than the response at `GET /foo` (we'll call this `Foo` vs FooLite`).

I would like to say that `FooLite` is always a subset of `Foo`, but that is not the case. In some situations, `FooLite` contains fields that are not present in `Foo`. I believe there are some examples of `Foo` and `FooLite` sharing a common field name with different response formats or values.

In short, they should be treated as related, but not necessarily the same nor extending.
